### DISCUSSION TITLE
Why most USB Hub doesn't support this feature? I can answer...

### DISCUSSION THREAD

**Discussion by KeithChao** [mvp/uhubctl#342] [Show and tell]
The value of bmRequestType of Table 11-15 Hub Class Request for Clear Port Feature is wrong with USB 2.0 to USB 3.2 specs.
ClearPortFeature should not be 00100011B. It has to change to 00100001B.
SetPortFeature is 00100011B.

Please send a mail to techsup@usb.org to fix this.

**Comment by mvp** [comment]
Extraordinary claims require extraordinary evidence. I don't understand why you think it is incorrect.

 Also, I don't understand what uhubctl has to do with this, and if it could improve anything regarding power switching.

Right now uhubctl works for quite a few USB hubs, from USB 1.1 all the way to USB 3.2. Granted, it would be nice if this list was larger, but so far I don't see any evidence of using wrong values for feature codes or selectors.

Even if that value you are talking about is incorrect, it is not likely that uhubctl can fix anything. What would probably involve fixing kernels, and probably USB hubs - good luck with that.

**Reply by KeithChao** [reply]
I feel embarrassing on my post. 

I failed to cycle the hub port power by software under Windows.
Try to get some solution from the Github code but still fail.

Eventfully, I changed my plan and used hardware to switch the VUSB by add R and MOS to the DIO7003(VUSB)-EN pin.
Use Arduino to control two hub 8 ports as a temporary solution.

**Reply by mvp** [reply]
According to Microsoft, USB power switching cannot be supported by any tool except by Windows kernel itself https://social.msdn.microsoft.com/Forums/sqlserver/en-US/f680b63f-ca4f-4e52-baa9-9e64f8eee101.
uhubctl actually compiles on Windows, it wouldn't work because of this problem.

That said, you could do it by attaching your devices or PPPS compliant USB hubs to Raspberry Pi and controlling it over the network from your Windows box.
